src/gmt_api.c

```diff
@@ -11027,6 +11027,12 @@ int GMT_Call_Module (void *V_API, const char *module, int mode, void *args) {
 	else {	/* Call the function and pass back its return value */
 		gmt_manage_workflow (API, GMT_USE_WORKFLOW, NULL);		/* First detect and set modern mode if modern mode session dir is found */
 		gmtapi_check_for_modern_oneliner (API, module, mode, args);	/* If a modern mode one-liner we must switch run--mode here */
+		if (API->external && gmt_M_is_verbose (API->GMT, GMT_MSG_INFORMATION)) {
```

Comment:
I don't see any messages. Here you use `GMT_MSG_INFORMATION`, but later you use `GMT_MSG_DEBUG`. Does it make sense?

Reply:
Makes no sense.  Cut/paste, now fixed.

Reply:
If no commands are printed, perhaps the verbosity level is not set yet? API->external is set to true if GMT_SESSION_EXTERNAL is passed to GMT_Create_Session,a nd I think you do that.  So the only other way to fail (assuming you set -Vi somehow) is that it is not really set?  Parsing of the command has not happened yet, so need a way to do that.

Reply:
Maybe try gmt set GMT_VERBOSE information before the command?

Reply:
Yes, `-Vi` doesn't work, but `gmt set GMT_VERBOSE` works.

However, it only outputs the arguments, not the module names.
```
pygmt-session [INFORMATION]: GMT_Call_Command string: 53c2e5b326d044899b91d46242b63ee3 -
pygmt-session [INFORMATION]: GMT_Call_Command string: -B -JM8i -R0/10/0/20 -Vi
pygmt-session [INFORMATION]: GMT_Call_Command string: 53c2e5b326d044899b91d46242b63ee3 -
pygmt-session [INFORMATION]: GMT_Call_Command string: @earth_relief_01d -Vi
pygmt-session [INFORMATION]: GMT_Call_Command string: 53c2e5b326d044899b91d46242b63ee3 -
pygmt-session [INFORMATION]: GMT_Call_Command string: -Ia/0.5,skyblue -Sskyblue -Vi
pygmt-session [INFORMATION]: GMT_Call_Command string: 53c2e5b326d044899b91d46242b63ee3 -
pygmt-session [INFORMATION]: GMT_Call_Command string: -N1/1,black -Vi
pygmt-session [INFORMATION]: GMT_Call_Command string: 53c2e5b326d044899b91d46242b63ee3 -
pygmt-session [INFORMATION]: GMT_Call_Command string: -A -Fales -Qg2 -Qt2 -Tg
pygmt-session [INFORMATION]: GMT_Call_Command string:
end [INFORMATION]: GMT_Call_Command string: 53c2e5b326d044899b91d46242b63ee3
```

Reply:
Just tested this branch, and it seems to work, though I wouldn't say the 'GMT_Call_Command' string is very readable:

<details>

```bash
grdimage [INFORMATION]: GMT_Call_Command string: ~/.gmt/sessions/gmt_session.116530/=tiled_80_PO.000000 -R3.966666666666667/4.270833333333333/44/44.25 -I03s -rg -G@GMTAPI@-S-O-G-G-G-Y-000002 -fg -Co+n -N0
grdblend [INFORMATION]: Reading Data Table from file ~/.gmt/sessions/gmt_session.116530/=tiled_80_PO.000000
grdblend [INFORMATION]: Given domain implies x_inc = 0.000833333
grdblend [INFORMATION]: Given domain implies y_inc = 0.000833333
grdblend [INFORMATION]: Round-off patrol changed grid limit for ymin from 40.00000000000001 to 40.00000000000001
grdblend [INFORMATION]: Round-off patrol changed grid limit for ymax from 50.00000000000002 to 50.00000000000002
grdblend [INFORMATION]: Blend file ~/.gmt/server/earth/earth_relief/earth_relief_03s_g/N44E003.earth_relief_03s_g.nc in 3/4/44/45 with normal weight 1 [-900-300]
grdblend [INFORMATION]: Blend file ~/.gmt/server/earth/earth_relief/earth_relief_03s_g/N44E004.earth_relief_03s_g.nc in 4/5/44/45 with normal weight 1 [-900-300]
grdblend [WARNING]: File @N40E000.earth_relief_15s_p.nc has different increments (0.00416666666667/0.00416666666667) than the output grid (0.000833333333333/0.000833333333333) - must resample
grdblend [WARNING]: File @N40E000.earth_relief_15s_p.nc coordinates are phase-shifted w.r.t. the output grid - must resample
grdblend [INFORMATION]: Resample @N40E000.earth_relief_15s_p.nc via grdsample @N40E000.earth_relief_15s_p.nc -rg -I0.000833333333333/0.000833333333333 -R3.96666666667/4.27083333333/44/44.25 -G/tmp/grdblend_resampled_116530_2.nc -Vi -fg --GMT_HISTORY=readonly
grdblend [INFORMATION]: GMT_Call_Command string: @N40E000.earth_relief_15s_p.nc -rg -I0.000833333333333/0.000833333333333 -R3.96666666667/4.27083333333/44/44.25 -G/tmp/grdblend_resampled_116530_2.nc -Vi -fg --GMT_HISTORY=readonly
grdsample [INFORMATION]: Processing input grid
grdsample [INFORMATION]: Round-off patrol changed grid limit for ymin from 40.00000000000001 to 40.00000000000001
grdsample [INFORMATION]: Round-off patrol changed grid limit for ymax from 50.00000000000002 to 50.00000000000002
grdsample [INFORMATION]: Given domain implies x_inc = 0.000833333
grdsample [INFORMATION]: Given domain implies y_inc = 0.000833333
grdsample [INFORMATION]: Input  grid (0/10/40/50) n_columns = 2400 n_rows = 2400 dx = 0.00416666666667 dy = 0.00416666666667 registration = 1
grdsample [INFORMATION]: Output grid (3.96666666667/4.27083333333/44/44.25) n_columns = 366 n_rows = 301 dx = 0.000833333333315 dy = 0.000833333333333 registration = 0
grdsample [INFORMATION]: Reading grid from file ~/.gmt/server/earth/earth_relief/earth_relief_15s_p/N40E000.earth_relief_15s_p.nc
grdsample [INFORMATION]: All boundaries set via extended data.
grdsample [INFORMATION]: Writing grid to file /tmp/grdblend_resampled_116530_2.nc
grdblend [INFORMATION]: Round-off patrol changed grid limit for xmin from 3.966666666579721 to 3.966666666579721
grdblend [INFORMATION]: Round-off patrol changed grid limit for xmax from 4.270833333239721 to 4.270833333239721
grdblend [INFORMATION]: Blend file /tmp/grdblend_resampled_116530_2.nc in 0/10/40/50 with normal weight 1 [0-300]
grdblend [INFORMATION]: Processing input grids
grdblend [INFORMATION]: Round-off patrol changed grid limit for xmin from 3.966666666579721 to 3.966666666579721
grdblend [INFORMATION]: Round-off patrol changed grid limit for xmax from 4.270833333239721 to 4.270833333239721
grdblend [INFORMATION]: Processed row       0 of 301
grdblend [INFORMATION]: Processed row      10 of 301
grdblend [INFORMATION]: Processed row      20 of 301
grdblend [INFORMATION]: Processed row      30 of 301
grdblend [INFORMATION]: Processed row      40 of 301
grdblend [INFORMATION]: Processed row      50 of 301
grdblend [INFORMATION]: Processed row      60 of 301
grdblend [INFORMATION]: Processed row      70 of 301
grdblend [INFORMATION]: Processed row      80 of 301
grdblend [INFORMATION]: Processed row      90 of 301
grdblend [INFORMATION]: Processed row     100 of 301
grdblend [INFORMATION]: Processed row     110 of 301
grdblend [INFORMATION]: Processed row     120 of 301
grdblend [INFORMATION]: Processed row     130 of 301
grdblend [INFORMATION]: Processed row     140 of 301
grdblend [INFORMATION]: Processed row     150 of 301
grdblend [INFORMATION]: Processed row     160 of 301
grdblend [INFORMATION]: Processed row     170 of 301
grdblend [INFORMATION]: Processed row     180 of 301
grdblend [INFORMATION]: Processed row     190 of 301
grdblend [INFORMATION]: Processed row     200 of 301
grdblend [INFORMATION]: Processed row     210 of 301
grdblend [INFORMATION]: Processed row     220 of 301
grdblend [INFORMATION]: Processed row     230 of 301
grdblend [INFORMATION]: Processed row     240 of 301
grdblend [INFORMATION]: Processed row     250 of 301
grdblend [INFORMATION]: Processed row     260 of 301
grdblend [INFORMATION]: Processed row     270 of 301
grdblend [INFORMATION]: Processed row     280 of 301
grdblend [INFORMATION]: Processed row     290 of 301
grdblend [INFORMATION]: Processed row     300 of 301
grdblend [INFORMATION]: Processed row     301
grdblend [INFORMATION]: Referencing grid data to GMT_GRID memory location
grdblend [INFORMATION]: Set boundary condition for all edges: natural
grdblend [INFORMATION]: Set boundary condition for left   edge: natural
grdblend [INFORMATION]: Set boundary condition for right  edge: natural
grdblend [INFORMATION]: Set boundary condition for bottom edge: natural
grdblend [INFORMATION]: Set boundary condition for top    edge: natural
grdblend [INFORMATION]: Blended grid size of @GMTAPI@-S-O-G-G-G-Y-000002 is 366 x 301
grdblend [INFORMATION]: All nodes assigned values
grdimage [INFORMATION]: Set boundary condition for all edges: natural
grdimage [INFORMATION]: Set boundary condition for left   edge: natural
grdimage [INFORMATION]: Set boundary condition for right  edge: natural
grdimage [INFORMATION]: Set boundary condition for bottom edge: natural
grdimage [INFORMATION]: Set boundary condition for top    edge: natural
grdimage [INFORMATION]: Central meridian not given, default to 4.12
grdimage [INFORMATION]: Map scale is 1.6435 km per cm or 1:164350.
grdimage [INFORMATION]: Derive intensity grid from data grid
grdimage [INFORMATION]: Calling grdgradient with args -G@GMTAPI@-S-O-G-G-G-Y-000015 -A-45.0 -Nt1+a0 -R3.966666666666667/4.270833333333333/44/44.25 --GMT_HISTORY=readonly @GMTAPI@-S-I-G-G-G-N-000001
grdimage [INFORMATION]: GMT_Call_Command string: -G@GMTAPI@-S-O-G-G-G-Y-000015 -A-45.0 -Nt1+a0 -R3.966666666666667/4.270833333333333/44/44.25 --GMT_HISTORY=readonly @GMTAPI@-S-I-G-G-G-N-000001
grdgradient [INFORMATION]: Processing input grid
grdgradient [INFORMATION]: Set boundary condition for all edges: natural
grdgradient [INFORMATION]: Set boundary condition for left   edge: natural
grdgradient [INFORMATION]: Set boundary condition for right  edge: natural
grdgradient [INFORMATION]: Set boundary condition for bottom edge: natural
grdgradient [INFORMATION]: Set boundary condition for top    edge: natural
grdgradient [INFORMATION]:  Min Mean Max sigma intensities:
grdgradient [INFORMATION]: 	-0.720555859751	-0.00787141662076	0.890532427816	0.126627940389
grdgradient [INFORMATION]: Referencing grid data to GMT_GRID memory location
grdgradient [INFORMATION]: Set boundary condition for all edges: natural
grdgradient [INFORMATION]: Set boundary condition for left   edge: natural
grdgradient [INFORMATION]: Set boundary condition for right  edge: natural
grdgradient [INFORMATION]: Set boundary condition for bottom edge: natural
grdgradient [INFORMATION]: Set boundary condition for top    edge: natural
grdimage [INFORMATION]: Allocate and read data from file ~/.gmt/sessions/gmt_session.116530/=tiled_80_PO.000000
grdimage [INFORMATION]: Allocates memory and read intensity file
grdimage [INFORMATION]: project grid files
grdimage [INFORMATION]: Grid projection from size 364x301 to 364x301
grdimage [INFORMATION]: gmt_grd_project: Output grid extrema [73/1068] exceed extrema of input grid [73/715] due to resampling
grdimage [INFORMATION]: gmt_grd_project: Output grid clipped to input grid extrema
grdimage [INFORMATION]: Grid projection from size 366x301 to 366x301
grdimage [INFORMATION]: Reading CPT from File ~/gmt-install-dir/share/cpt/geo.cpt
grdimage [INFORMATION]: Auto-stretching CPT file geo to fit data range 73 to 715
grdimage [INFORMATION]: gmtsupport_validate_cpt: CPT hard hinge is outside actual data range - range adjusted to start at hinge 73 and below-hinge CPT ignored.
grdimage [INFORMATION]: Write CPT to File ~/.gmt/sessions/gmt_session.116530/gmt.2.cpt
grdimage [INFORMATION]: Save current CPT file to ~/.gmt/sessions/gmt_session.116530/gmt.2.cpt !
grdimage [INFORMATION]: Evaluate image pixel colors
grdimage [INFORMATION]: Basic z(x,y) with optional illumination and no PostScript colormasking.
grdimage [INFORMATION]: Creating 24-bit color image
PSL: Too many colors to make colormap - using 24-bit direct color instead.
PSL: DEFLATE compressed 328692 to 312437 bytes (4.9% savings at compression level 5)
```


</details>